# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
# Commit: 72d4bf5b093e
# Full Hash: 72d4bf5b093e9b00eed78bf4e51850d56df5ae6c
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-05 16:09:03
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add RenderTextureHostSWGL support to D3D11 RenderTextureHosts. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95799
# ==============================================================================

diff -r 99b2e5c55a25 -r 72d4bf5b093e gfx/webrender_bindings/RenderCompositorD3D11SWGL.h
--- a/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Thu Nov 05 09:14:52 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositorD3D11SWGL.h	Thu Nov 05 09:15:05 2020 +0000
@@ -49,6 +49,7 @@
   layers::WebRenderCompositor CompositorType() const override {
     return layers::WebRenderCompositor::D3D11;
   }
+  RenderCompositorD3D11SWGL* AsRenderCompositorD3D11SWGL() { return this; }
 
   // Interface for wr::Compositor
   CompositorCapabilities GetCompositorCapabilities() override;
@@ -95,6 +96,8 @@
     int32_t mY;
   };
 
+  ID3D11Device* GetDevice() { return mCompositor->GetDevice(); }
+
  private:
   RefPtr<layers::CompositorD3D11> mCompositor;
   void* mContext = nullptr;